src/ape_compile/_cli.py

```diff
@@ -26,28 +33,53 @@ def cli(contracts, compile_all, display_size):
     a project is loaded. You do not have to manually trigger a recompile.
     """
 
-    # TODO
-    # * Look for local manifest
-    # * If not, look for local config
-    # * If no config assume everything is default
-    # * `contracts` allows selecting specific contracts to compile from manifest
-    # Config should override defaults. If no config value, use default.
-    # default config is the source of truth, the manifest is an artifact of the
-    # config, both pre-and post procesing
-
-    if len(project.sources) == 0:
-        notify("ERROR", "Manifest contains no sources to compile")
+    # Expand source tree based on selection
+    if not filepaths:
+        if not (Path.cwd() / "contracts").exists():
+            notify("ERROR", "No `contracts/` directory detected")
+            return
+
+        # If no paths are specified, use all local project sources
+        contract_filepaths = project.sources
+
+    else:
+        # Turn all paths into `pathlib.Path` objects
+        contract_filepaths = []
+        for path in map(Path, filepaths):
+            if path.is_dir():
+                contract_filepaths.extend(list(path.iterdir()))
+
+            else:
+                contract_filepaths.append(path)
+
+        # Filter by what's in our project's source tree
+        contract_filepaths = [c for c in contract_filepaths if c in project.sources]
+
+    if len(contract_filepaths) == 0:
```

Comment:
I think it's slightly more idiomatic python to say `if not contract_filepaths` (same as what you did on line 37).

Reply:
```suggestion
    if not contract_filepaths:
```